packages/odyssey-react-mui/src/labs/OdysseyPickers/ComposablePicker.tsx

```diff
@@ -0,0 +1,191 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { memo, ReactNode } from "react";
+import {
+  Autocomplete as MuiAutocomplete,
+  AutocompleteProps as MuiAutocompleteProps,
+} from "@mui/material";
+
+import { AutocompleteProps } from "../../Autocomplete";
+import { useAutocomplete } from "../../useAutocomplete";
+
+export const adornmentSizeValues = ["small", "large"] as const;
+export type AdornmentSize = (typeof adornmentSizeValues)[number];
+
+export type BasePickerProps<
+  OptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined,
+> = AutocompleteProps<OptionType, HasMultipleChoices, IsCustomValueAllowed> & {
+  adornmentSize?: AdornmentSize;
+  groupOptionsBy?: MuiAutocompleteProps<
+    OptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["groupBy"];
+};
+
+export type BasePickerType = {
+  displayName: string;
+  <
+    OptionType,
+    HasMultipleChoices extends boolean | undefined,
+    IsCustomValueAllowed extends boolean | undefined,
+  >(
+    props: BasePickerProps<
+      OptionType,
+      HasMultipleChoices,
+      IsCustomValueAllowed
+    >,
+  ): ReactNode;
+};
+
+export type ComposablePickerProps<
+  OptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined,
+> = BasePickerProps<OptionType, HasMultipleChoices, IsCustomValueAllowed> & {
+  getOptionLabel?: MuiAutocompleteProps<
+    OptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["getOptionLabel"];
+
+  renderOption: MuiAutocompleteProps<
+    OptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["renderOption"];
+
+  renderTags?: MuiAutocompleteProps<
+    OptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["renderTags"];
+};
+
+const ComposablePicker = <
+  OptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined,
+>({
+  ariaDescribedBy,
+  defaultValue,
+  errorMessage,
+  errorMessageList,
+  getIsOptionEqualToValue,
+  getOptionLabel,
+  groupOptionsBy,
+  hasMultipleChoices,
+  id: idOverride,
+  inputValue,
+  isCustomValueAllowed,
+  isDisabled,
+  isFullWidth = false,
+  isLoading,
+  isOptional = false,
+  isReadOnly,
+  isVirtualized: isVirtualizedProp = false,
+  hint,
+  HintLinkComponent,
+  label,
+  name: nameOverride,
+  onBlur,
+  onChange: onChangeProp,
+  onInputChange: onInputChangeProp,
+  onFocus,
+  options,
+  renderOption,
+  renderTags,
+  value,
+  testId,
+  translate,
+}: ComposablePickerProps<
+  OptionType,
+  HasMultipleChoices,
+  IsCustomValueAllowed
+>) => {
+  const {
+    inputValueProp,
+    isVirtualized,
```

Comment:
If this is a ref, we should name it `isVirtualizedRef` just so it's clear it has a `.current`.